### ISSUE TITLE
auto-fill-mode: too aggressive?

### ISSUE THREAD

**Issue by ixtenu** [japanoise/gomacs#30]
When `auto-fill-mode` is enabled, Gomacs calls `doFillParagraph` whenever space is pressed and the current line exceeds the fill column.  Currently, https://github.com/japanoise/gomacs/issues/29 makes that behavior quite annoying when inserting text into the middle of a paragraph, because you insert a few characters, press space, and suddenly point is at the end of the paragraph.

GNU Emacs (and most other Emacsen with `auto-fill-mode`) has a less aggressive implementation which only fills the current line (as opposed to the whole paragraph) when space is pressed and _point_ (as opposed to the end of line) exceeds the fill column.  To fill the whole paragraph, `M-q` must be explicitly used.

It wouldn't be hard to do the same in Gomacs.  If https://github.com/japanoise/gomacs/issues/29 is _not_ resolved, then I would restrict it further, so that `doFillParagraph` also doesn't occur unless point is at the end of line; this is what my [auto-fill-fixes-wip](https://github.com/japanoise/gomacs/compare/master...ixtenu:auto-fill-fixes-wip) branch implemented.  Though it might be better to deal with https://github.com/japanoise/gomacs/issues/29 first.  Even with that issue fixed, I think a less aggressive `auto-fill-mode` is warranted, since even in GNU Emacs (which has a fairly fancy implementation of `fill-paragraph`) there are times when invoking `fill-paragraph` is undesired but filling the current line is fine.

(This could also be made configurable.  Vim has `set formatoptions+=a` which enables something similar to what I've been calling "aggressive" `auto-fill-mode`.)

**Comment by japanoise** [comment]
Yes, this is a reasonable change. We can maintain the existing (aggressive) behavior with a `aggressive-fill-mode` or somesuch, since sometimes it's desirable. Let's take care of #29 first.

**Comment by japanoise** [comment]
With #29 finished and the other PRs taken care of, now would be a good time to get this taken care of, if you have time @ixtenu
